wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/NamespaceResolutionContext.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.server.spi;
+
+import java.util.Objects;
+
+/**
+ * Infrastructure may be configured to use namespace value that should be evaluated. This objects
+ * holds everything that might be needed for such evaluation.
+ *
+ * @author Sergii Leshchenko
+ */
+public class NamespaceResolutionContext {
+  private final String workspaceId;
+  private final String userId;
+  private final String userName;
+
+  public NamespaceResolutionContext(String workspaceId, String userId, String userName) {
+    this.workspaceId = workspaceId;
+    this.userId = userId;
+    this.userName = userName;
+  }
+
+  public String getWorkspaceId() {
+    return workspaceId;
+  }
+
+  public String getUserId() {
+    return userId;
+  }
+
+  public String getUserName() {
+    return userName;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (!(obj instanceof NamespaceResolutionContext)) {
+      return false;
+    }
+    final NamespaceResolutionContext that = (NamespaceResolutionContext) obj;
+    return Objects.equals(workspaceId, that.workspaceId)
+        && Objects.equals(userId, that.userId)
+        && Objects.equals(userName, that.userName);
+  }
+
+  @Override
+  public int hashCode() {
+    int hash = 7;
+    hash = 31 * hash + Objects.hashCode(workspaceId);
+    hash = 31 * hash + Objects.hashCode(userId);
+    hash = 31 * hash + Objects.hashCode(userName);
+    return hash;
+  }
+
+  @Override
+  public String toString() {
+    return "PlaceholderResolutionContext{"
```

Comment:
You've been caught plagiarizing :smile:

Reply:
I did not try to hide it )
Thank IDE refactoring tools )